includes/derivatives.inc

```diff
@@ -296,72 +320,37 @@ function islandora_ocr_derive_ocr(AbstractObject $object, $force = FALSE) {
+  return islandora_ocr_derive_datastream('OCR', ISLANDORA_OCR_GENERATE_OCR_RELS_STATEMENT, $object, $force);
+}
+
+/**
  * Derive the HOCR stream from OBJ.
  */
 function islandora_ocr_derive_hocr(AbstractObject $object, $force = FALSE) {
-  $language_array = $object->relationships->get(ISLANDORA_RELS_EXT_URI, 'hasLanguage');
-  // Logic for HOCR generation from islandora_book_batch.
-  $generate_hocr = 'TRUE';
-  $batch_hocr = $object->relationships->get(ISLANDORA_RELS_EXT_URI, 'generate_hocr');
-  if ($batch_hocr) {
-    $generate_hocr = $batch_hocr[0]['object']['value'];
-  }
-  $language = isset($language_array[0]['object']['value']) ? $language_array[0]['object']['value'] : 'eng';
-  if ($language == 'no_ocr' || $generate_hocr == 'FALSE') {
-    $to_return['success'] = TRUE;
-    $to_return['messages'][] = array(
-      'message' => t('Skipped adding @dsid to @id.'),
-      'message_sub' => array(
-        '@dsid' => 'HOCR',
-        '@id' => $object->id,
-      ),
-      'type' => 'dsm',
-      'severity' => 'status',
-    );
-    return $to_return;
+  // If this object is older than the time this module was updated to 7001, we
+  // want to use the OCR flag instead of the HOCR flag to determine if we should
+  // derive. If no update was required, no check is either.
+  $timestamp = variable_get(ISLANDORA_OCR_UPDATE_7001_TIMESTAMP, NULL);
+  if (!isset($timestamp)) {
+    $flag = ISLANDORA_OCR_GENERATE_HOCR_RELS_STATEMENT;
   }
-  if ($generate_hocr == 'TRUE' && ($force || !isset($object['HOCR']))) {
-    $to_return = array(
-      'success' => FALSE,
-      'messages' => array(),
-    );
-
-    $datastream = $object['OBJ'];
-    $file = islandora_ocr_get_converted_source($datastream);
-    $ocr_file = islandora_ocr_create_hocr($file, array('language' => $language));
-    if ($ocr_file && islandora_ocr_update_datastream($object, $ocr_file, 'HOCR')) {
-      $to_return['success'] = TRUE;
-      $to_return['messages'][] = array(
-        'message' => t('Added @dsid to @id.'),
-        'message_sub' => array(
-          '@dsid' => 'HOCR',
-          '@id' => $object->id,
-        ),
-        'type' => 'dsm',
-        'severity' => 'status',
-      );
-    }
-    else {
-      $to_return['messages'][] = array(
-        'message' => t('Failed to generate @dest_dsid from @src_dsid for @id.'),
-        'message_sub' => array(
-          '@dest_dsid' => 'OCR',
-          '@src_dsid' => 'OBJ',
-          '@id' => $object->id,
-        ),
-        'type' => 'watchdog',
-        'severity' => WATCHDOG_ERROR,
-      );
-    }
-    // Clean up.
-    file_unmanaged_delete($file);
-    return $to_return;
+  else {
+    $diff = date_diff(new DateTime($timestamp), $object->createdDate);
+    // Negative diffs indicate an object created before the timestamp.
+    $flag = $diff->invert ?
+      ISLANDORA_OCR_GENERATE_OCR_RELS_STATEMENT :
+      ISLANDORA_OCR_GENERATE_HOCR_RELS_STATEMENT;
```

Comment:
Would probably be safer just to make it copy the value over, instead of trying to cross wires... Though... not sure where would be the best place... Would have to be where the OCR one is sure to still exist...